### DISCUSSION TITLE
깃허브 기본 트레이닝/LAB

### DISCUSSION THREAD

**Discussion by bryantson** [Korean-American-IT-Association-KAITA/kaita-project-management#19] [5. 트레이닝/멘토링 이벤트]
기본적인 Git/GitHub 트레이닝입니다.

가능한 시간이나 관심있는 분들은 댓글 달아주세요

**Comment by bbkaeul** [comment]
깃허브 트레이닝 필요합니다. 
가능한 시간: 6:00CST -24:00 CST

**Comment by j2jun** [comment]
깃허브 트레이닝 필요합니다.
가능한 시간: 금, 토 8pm-11pm cst

**Comment by yenny1124** [comment]
깃허브 트레이닝 필요합니다
가능한 시간: 금,토,일 CST 2pm ~ 이후 아무때나

**Comment by ndk6879** [comment]
깃허브 트레이닝 필요합니다
가능한 시간: 토 CST 7pm ~ 이후 아무때나

**Comment by ghkfehd0502** [comment]
깃허브 트레이닝 필요합니다.
가능한 시간: 5pm CST 이후

**Comment by judyyhham** [comment]
깃허브 트레이닝 필요합니다
가능한 시간: 월, 화, 금, 토 6pm CST 이후

**Comment by bryantson** [comment]
토욜 7PM CST 이후면 다 가능할것 같네여

**Comment by bryantson** [comment]
이번 깃허브 트레이닝은 여기서 마감하겠습니다. 시간은 토욜 7 PM CST 나 이후로 잡아보겠습니다

**Comment by bryantson** [comment]
프로젝트 메니지먼트 트레이닝은 10/09/21 토욜 7PM CST - 8 PM CST그리고 깃허브 트레이닝은 8:30PM CST - 9:30 PM CST 생각합니다

**Comment by bryantson** [comment]
금요일 8 PM CST 가 나을것 같네요

**Reply by Ssuny** [reply]
참여하고 싶지만 스케줄이 안맞아 참가할수 없는 분들을 위해 영상 녹화를 해놓는건 어떨까요?

**Reply by bryantson** [reply]
그럴수도 있지만 LAB 이라. 한번 동의를 얻고 해보겠습니다.

**Comment by bryantson** [comment]
KAITA 프로젝트에 필요한 깃허브 기본 트레이닝 입니다. 랩 식으로 진행 될 예정 입니다. 오시기 전에 랩탑 준비하시고 Visual Studio Code 같은 text editor 준비 하시길 바랍니다.


1. 마크 다운 (Markdown) 파일 - 설명
2. 마크 다운 돌아가면서 직접 해보기
3. 브랜치 개념 설명
4. 브랜치 돌아가면서 직접 해보기
5. 풀리퀘스트 + 코드 리뷰 설명 
6. 풀리퀘스 직접 해보기
7, 그 외

**Comment by Eunyoungkim0** [comment]
금요일 8 PM CST 가능합니다!

**Comment by bryantson** [comment]
@Silverstelle 

더 했습니다

**Reply by Eunyoungkim0** [reply]
감사합니당

**Comment by bryantson** [comment]
LAB 에 참여 하시는 분들 기본적으로 몇가지 준비 하셔야 합니다

1. 깃허브 어카운트 로그인 준비된 웹브라우져
2. 화면 쉐어 기능 가능할 환경 
3. Visual Studio Code 같은 text editor 미리 다운로드나 인스톨
4. Git 다운해서 인스톨 하면 좋음

**Reply by Ssuny** [reply]
오늘 미팅에 참여 가능할것 같습니다.
invitation 보내주실수 있나요?

**Comment by kikibell1004** [comment]
안녕하세요 
깃허브 트레이닝 참석하고 싶습니다. 
invitation 보내주실 수 있으세요?

**Comment by Ssuny** [comment]
감사합니다.

**Comment by bryantson** [comment]
깃허브 첫번째 트레이닝에 참여해 주신 분들께 감사 드립니다. 다음 트레이닝은 필요에 따라 Ad-Hoc 식으로 하겠습니다. 복습을 위해서 숙제를 하나 준비 해봤습니다.

## 깃허브 숙제 1

* 모든 프로세스는 **main** 브랜치가 아닌 각자의 브랜치에서 일하고 add > commit > push  > pull request  순서로 하셔야 합니다.

- [ ] **이미 만들어진 파일 클린업**: 저희가 이 레포 **kaita-project-management** 안에 **src** 폴더가 있고 그 안에 각자의 브랜치를 만들어서 파이썬 파일을 만드는 과정을 해봤습니다. 이제 그 본인 브랜치로 바꾼다음 파일을 지우고 브랜치도 지우세요. 브랜치는 본인 컴터인 local 에서도 없어져야 하고 리모트에서도 없어져야 합니다. 지우는건 **Pull Request**  그리고 **리뷰를** 를 통해서 지우세요.
- [ ] **좀 더 디테일한 프로필 만들기**: 저희가 [깃허브 디스커션을 통해서 간단한 소개](https://github.com/Korean-American-IT-Association-KAITA/kaita-project-management/discussions/17) 는 했지만 자세한 소개는 하지 못했습니다.  **kaita-project-management** 에 새로운 디렉토리 **members** 를 만들고 **본인 깃허브 아이디-영어 이름.md** (예: bryantson-Bryant-Son) 식으로 좀 더 자세한 소개 프로필을 만들어주세요. 프로필에는 자신 있는 테크놀로지, 취미, 특별한 점들 등등 좀 더 자유롭게 적어 주시면 됩니다.
- [ ] **README.md 더 질서 있게 꾸미기:** 저희 **kaita-project-management** 레포에 **README.md** 파일이 있지만 아직 정리가 잘 안되어 있습니다. 좋은 다큐멘테이션은 아주 중요하고 깃허브 레포에선 이런 마크다운 파일이 그 역할을 합니다. 그래서 [여기 있는 다양한 깃허브 마크다운이 잘 되어 있는 레포를 참고해서](https://github.com/matiassingers/awesome-readme) 저희 **kaita-project-management** 레포에 **README.md** 을 좀 더 유용하고 깔끔하게 만들어 주세요. 한글로 되어 있어야 하고 이 레포가 뭘하는 건지 저희가 할려는 목적인 어떤건지 설명이 되어 있으면 좋겠습니다.

### 추가 설명
각자에게 이 숙제를 GitHub Issue 로 assign 해서 보내겠습니다. 궁금한점 있으면 [디스커션 보드 Q&A](https://github.com/Korean-American-IT-Association-KAITA/kaita-project-management/discussions/categories/4-q-a) 를 사용하는걸 추천합니다.

### 유용한 링크 & 레퍼런스
- [다양한 깃허브 마크다운이 잘 되어 있는 레포들 예](https://github.com/matiassingers/awesome-readme)
- [깃허브 마크 다운](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)
- [마크 다운 가이드](https://guides.github.com/features/mastering-markdown/) 
- [깃허브 마크다운 이모지](https://www.webfx.com/tools/emoji-cheat-sheet/)
